types/networks/src/lib.rs

```diff
@@ -236,14 +238,11 @@ impl ChainConfig {
             .map(|info| info.height)
             .unwrap_or(Height::Breeze);
 
-        let version = self
-            .version_schedule
+        self.version_schedule
             .iter()
             .find(|info| height == info.height)
             .map(|info| info.version)
-            .unwrap();
-
-        NetworkVersion::try_from(version).unwrap()
+            .unwrap()
```

Comment:
I'd always try to avoid panics by either providing a default or formulating an expectation that helps to identify panics later.

```suggestion
            .unwrap_or(NetworkVersion::V1)
```

Reply:
makes sense, the unwrap() initially was assumed because a network height should exist even if not specified in the config (a default exists). The redundancy makes sense here as a just in case. done

Reply:
Yes, I saw that. I just wanted to discourage the use of a plain `unwrap()`s. In that case the following would alos look good:

```suggestion
            .expect("A network height should exist even if not specified in the config (a default exists).")
```

Reply:
That make sense. I'll commit your suggestion and it should be g2g

Reply:
Good remark but I wouldn't apply it here. `.except` with an helpinh message for the dev or node operator should be used instead here.
This should panics because it means you forget to add an `UpgradeInfo` entry in the `version_schedule` (code or .toml).
e.g. some epoch leads to `Height::Skyr` but you forget to add the entry `{ height = "Skyr", version = "V16" }`.